merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/AutoValueMappers.java

```diff
@@ -153,6 +188,123 @@ record ComponentMapperNamePair(String componentName, String mapperName) {}
               .build());
     }
 
+    record Property(String name, TypeMirror type, String mapperName) {}
+    for (final var annotation : annotationTypes) {
+      final var methodName = ClassName.get((TypeElement) annotation).canonicalName().replace(".", "_");
+      final var properties = new ArrayList<Property>();
+      final var necessaryMappers = new HashMap<TypeMirror, String>();
+      for (final var element : annotation.getEnclosedElements()) {
+        if (!(element.getKind() == ElementKind.METHOD)) continue;
+        final var typeMirror = ((ExecutableElement) element).getReturnType();
+        final var elementName = element.toString().replace("(", "").replace(")", "");
+        final var valueMapperIdentifier = JAVA_IDENTIFIER_ILLEGAL_CHARACTERS
+                                              .matcher(typeMirror.toString())
+                                              .replaceAll("_")
+                                          + "ValueMapper";
+        properties.add(new Property(elementName, typeMirror, valueMapperIdentifier));
+        necessaryMappers.put(typeMirror, valueMapperIdentifier);
+      }
+
+      builder.addMethod(
+          MethodSpec
+              .methodBuilder(methodName)
+              .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+              .returns(ParameterizedTypeName.get(ClassName.get(ValueMapper.class), TypeName.get(annotation.asType())))
+              .addParameters(
+                  necessaryMappers
+                      .entrySet()
+                      .stream()
+                      .map(mapperRequest -> ParameterSpec
+                          .builder(
+                              ParameterizedTypeName.get(
+                                  ClassName.get(ValueMapper.class),
+                                  ClassName.get(mapperRequest.getKey()).box()),
+                              mapperRequest.getValue(),
+                              Modifier.FINAL)
+                          .build())
+                      .toList())
+              .addCode(
+                  CodeBlock
+                      .builder()
+                      .add("return $L;\n",
+                           TypeSpec
+                               .anonymousClassBuilder(CodeBlock.of(""))
+                               .addSuperinterface(
+                                   ParameterizedTypeName.get(ClassName.get(ValueMapper.class), TypeName.get(annotation.asType())))
+                               .addMethod(
+                                   MethodSpec
+                                       .methodBuilder("getValueSchema")
+                                       .addModifiers(Modifier.PUBLIC)
+                                       .returns(ValueSchema.class)
+                                       .addStatement("return $T.STRING", ValueSchema.class)
```

Comment:
Bug on this line

Reply:
Great catch - I'll push the fix in a moment